common/app/routes/Settings/redux/update-user-epic.js

```diff
@@ -27,25 +26,37 @@ const urlMap = {
 };
 
 export function updateUserEmailEpic(actions, { getState }) {
-  return actions::ofType(types.updateMyEmail)
-    .flatMap(({ payload: email }) => {
-      const {
-        app: { user: username, csrfToken: _csrf },
-        entities: { user: userMap }
-      } = getState();
-      const { email: oldEmail } = userMap[username] || {};
+  const updateEmail = actions
+    .filter(({ type, payload }) => (
+      type === types.updateMyEmail && payload
+    ))
+    .map(({ payload }) => {
+      const state = getState();
+      const { email } = userSelector(state);
+      return { email: payload, oldEmail: email };
+    });
+  const ajaxUpdate = updateEmail
+    .debounce(250)
+    .flatMap(({ email, oldEmail }) => {
+      const { app: { user: username, csrfToken: _csrf } } = getState();
       const body = { _csrf, email };
-      const optimisticUpdate = Observable.just(
-        updateUserEmail(username, email)
-      );
-      const ajaxUpdate = postJSON$('/update-my-email', body)
-        .map(({ message }) => makeToast({ message }))
-        .catch(doActionOnError(() => oldEmail ?
-          updateUserFlag(username, oldEmail) :
-          null
-        ));
-      return Observable.merge(optimisticUpdate, ajaxUpdate);
+      return postJSON$('/update-my-email', body)
+        .flatMap(({ message }) => {
+          return Observable.of(
+            // show user that we have updated their email
+            makeToast({ message })
+          );
+        })
```

Comment:
Since flatmap here is only maping to a single action it can be swapped for a map

`map(makeToast)`